components/scream/src/physics/p3/p3_ic_cases.cpp

```diff
@@ -8,104 +8,110 @@ namespace p3 {
-
-  // input variables.
-  d.dt = 1800;
-
-  // compute vertical grid spacing dzq (in m) from pres and theta.
-  static constexpr double
-    g = 9.8; // gravity, m/s^2
-  for (k = 0; k < nk; ++k) {
-    double plo, phi; // pressure at cell edges, Pa
-    plo = (k == 0   ) ?
-      std::max<double>(0, d.pres(0,0) - 0.5*(d.pres(0,1) - d.pres(0,0))/(1 - 0)) :
-      0.5*(d.pres(0,k-1) + d.pres(0,k));
-    phi = (k == nk-1) ?
-      d.pres(0,nk-1) + 0.5*(d.pres(0,nk-1) - d.pres(0,nk-2))/(1 - 0) :
-      0.5*(d.pres(0,k) + d.pres(0,k+1));
-    const auto dpres = phi - plo;
-    d.dzq(0,k) = Constants<double>::RD*T(k)/(g*d.pres(0,k))*dpres;
+  for (Int i = 0; i < ncol; ++i) {
+    // For column i = 0, use the ICs as originally coded in python and
+    // subsequently modified here. For columns i > 0, introduce some small
+    // variations.
+
+    // max cld at ~700mb, decreasing to 0 at 900mb.
+    for (k = 0; k < 15; ++k) d.qc(i,nk-20+k) = 1e-4*(1 - double(k)/14);
+    for (k = 0; k < nk; ++k) d.nc(i,k) = 1e6;
+    // max rain at 700mb, decreasing to zero at surf.
+    for (k = 0; k < 20; ++k) d.qr(i,nk-20+k) = 1e-5*(1 - double(k)/19);
+    for (k = 0; k < nk; ++k) d.nr(i,k) = 1e6;
+
+    //                                                      v (in the python)
+    for (k = 0; k < 15; ++k) d.qitot(i,nk-20+k) = 1e-4; //*(1 - double(k)/14)
+    for (k = 0; k < nk; ++k) d.nitot(i,k) = 1e6;
+    for (k = 0; k < 15; ++k) d.qirim(i,nk-20+k) = 1e-4*(1 - double(k)/14);
+    // guess at reasonable value based on: m3/kg is 1/density and liquid water has
+    // a density of 1000 kg/m3
+    for (k = 0; k < 15; ++k) d.birim(i,nk-20+k) = 1e-2;
+
+    // qv goes to zero halfway through profile (to avoid condensate near model
+    // top)
+    for (k = 0; k < nk; ++k) {
+      const auto tmp = -5e-4 + 1e-3/double(nk)*k;
+      d.qv(i,k) = tmp > 0 ? tmp : 0;
+    }
+    // make layer with qc saturated.
+    for (k = 0; k < 15; ++k) d.qv(i,nk-20+k) = 5e-3;
+
+    // pres is actually an input variable, but needed here to compute theta.
+    for (k = 0; k < nk; ++k) d.pres(i,k) = 100 + 1e5/double(nk)*k;
+    // pdel is actually an input variable, but needed here to compute theta.
+    for (k = 0; k < nk; ++k) d.pdel(i,k) = 1e5/double(nk);
+    // exner is actually an input variable, but needed here to compute theta.
+    for (k = 0; k < nk; ++k) d.exner(i,k) = std::pow((1e5/d.pres(i,k)), (287.15/1005.0));
+    // cloud fraction is an input variable, just set to 1 everywhere
+    for (k = 0; k < nk; ++k) d.icldm(i,k) = 1.0;
+    for (k = 0; k < nk; ++k) d.lcldm(i,k) = 1.0;
+    for (k = 0; k < nk; ++k) d.rcldm(i,k) = 1.0;
+
+    // To get potential temperature, start by making absolute temperature vary
+    // between 150K at top of atmos and 300k at surface, then convert to potential
+    // temp.
+    FortranData::Array1 T("T", nk);
+    for (k = 0; k < nk; ++k) {
+      T(k) = 150 + 150/double(nk)*k;
+      if (i > 0) T(k) += ((i % 3) - 0.5)/double(nk)*k;
```

Comment:
This is the only meaningful change. Everything else in this function is simply converting `(0,k)` to `(i,k)` and indenting.